pandas/io/sas/sasreader.py

```diff
@@ -1,16 +1,24 @@
 """
 Read SAS sas7bdat or xport files.
 """
+from typing import TYPE_CHECKING, AnyStr, Optional, Union
+
+from pandas._typing import FilePathOrBuffer
+
 from pandas.io.common import stringify_path
 
+if TYPE_CHECKING:
+    from pandas.io.sas.sas_xport import XportReader  # noqa: F401
+    from pandas.io.sas.sas7bdat import SAS7BDATReader  # noqa: F401
+
 
 def read_sas(
-    filepath_or_buffer,
-    format=None,
+    filepath_or_buffer: FilePathOrBuffer[AnyStr],
```

Comment:
maybe we should change the definition in _typing?

Reply:
can you elaborate?

Reply:
FilePathOrBuffer[AnyStr] should be in _typing instead of just the not parametric FilePathOrBuffer

Reply:
for some read/write methods only a string buffer (or bytes buffer) is allowed, i.e. to_html and we need to annotate as FilePathOrBuffer[str] (or FilePathOrBuffer[bytes]) otherwise we get mypy errors.

Reply:
that’s a very limited case compared to the more general which accepts strings and buffers

these should be distinct / parametrized 

not just in this PR - this is my point

Reply:
not sure I understand, AnyStr is a TypeVar and therefore parametrized https://docs.python.org/3/library/typing.html#typing.AnyStr

Reply:
so, you are adding this here, rather than as a generailized soln in _typing. I would think it would be better there than adding to *each* of the IO readers.

Reply:
i reverted these additions for now to keep this PR focused on the untyped_defs error, https://github.com/pandas-dev/pandas/pull/30659#issue-359115419